src/exif.cpp

```diff
@@ -230,7 +230,19 @@ namespace Exiv2 {
               fct=NULL;
             }
         }
-        if ( fct ) fct(os, value(), pMetadata);
+        if ( fct ) {
+          // https://github.com/Exiv2/exiv2/issues/1706
+          // Sometimes the type of the value doesn't match what the
+          // print function expects. (The expected types are stored
+          // in the TagInfo tables, but they are not enforced when the
+          // metadata is parsed.) These type mismatches can sometimes
+          // cause a std::out_of_range exception to be thrown.
+          try {
+            fct(os, value(), pMetadata);
+          } catch (std::out_of_range&) {
+            os << "Bad value";
```

Comment:
I'm wondering if this should maybe additionally result in a warning to `stderr` ?

Reply:
I have added this now. It only prints when `EXIV2_DEBUG_MESSAGES` is enabled though.